staging/src/k8s.io/apiserver/pkg/util/peerproxy/peerproxy_handler.go

```diff
@@ -93,12 +93,11 @@ func (h *peerProxyHandler) HasFinishedSync() bool {
 }
 
 func (h *peerProxyHandler) WaitForCacheSync(stopCh <-chan struct{}) error {
-
-	ok := cache.WaitForNamedCacheSync("unknown-version-proxy", stopCh, h.storageversionInformer.HasSynced, h.storageversionManager.Completed)
+	h.startLocalDiscoveryCacheInvalidation(stopCh)
```

Comment:
nit: it's a bit hard to tell that this is not blocking since the goroutine is started within the function. Can we call `go h.startLocalDiscoveryCacheInvalidation(stopCh)` here and move the goroutine invocation up one layer?

Reply:
Moved some things here, does it look better now?